exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/join/TestMergeJoinAdvanced.java

```diff
@@ -255,6 +260,42 @@ public void testDrill4196() throws Exception {
       .go();
   }
 
+  @Test // DRILL-6491
+  public void testMergeIsNotSelectedForFullJoin() throws Exception {
+    setSessionOption(PlannerSettings.HASHJOIN.getOptionName(), true);
+
+    String query = "select * " +
+        " from (select employee_id from cp.`employee.json` order by employee_id) e1 " +
+        " full outer join (select employee_id from cp.`employee.json` order by employee_id) e2 " +
+        " on e1.employee_id = e2.employee_id " +
+        " limit 10";
+    try {
+      testPlanMatchingPatterns(query, new String[] {"HashJoin"}, new String[] {"MergeJoin"});
+    } finally {
+      test("ALTER SESSION RESET ALL");
```

Comment:
Reset only options, you have set.

Reply:
What if default values for the (explicitly) set options are changed in the future?

Reply:
Use ` resetSessionOption` method.